GeoLib/AnalyticalGeometry.cpp

```diff
@@ -514,21 +509,30 @@ bool isCoplanar(const MathLib::Point3d& a, const MathLib::Point3d& b,
 void computeAndInsertAllIntersectionPoints(GeoLib::PointVec &pnt_vec,
 	std::vector<GeoLib::Polyline*> & plys)
 {
+	auto computeSegmentIntersections = [&pnt_vec](GeoLib::Polyline& poly0,
+	                                              GeoLib::Polyline& poly1)
+	{
+		for (auto seg0_it(poly0.begin()); seg0_it != poly0.end(); ++seg0_it)
+		{
+			for (auto seg1_it(poly1.begin()); seg1_it != poly1.end(); ++seg1_it)
+			{
+				GeoLib::Point s(0.0, 0.0, 0.0, pnt_vec.size());
+				if (lineSegmentIntersect(*seg0_it, *seg1_it, s))
+				{
+					std::size_t const id(
+					    pnt_vec.push_back(new GeoLib::Point(s)));
+					poly0.insertPoint(seg0_it.getSegmentNumber() + 1, id);
```

Comment:
Will that mess up the segment numbers stored in the iterator when `seg0_it` is incremented afterwards?